### ISSUE TITLE
Godot 4.0: Tilemap: use_texture_padding does not (fully) work on hexagonal tiles

### ISSUE THREAD

**Issue by seppoday** [godotengine/godot#71160]
### Godot version

4.0 beta 7

### System information

Windows 10

### Issue description

I have an issue with edges in hexagonal tilemaps. I've tried changing filtering and other stuff. I've also tried new option in godot 4.0 -> use_texture_padding and it works fine but only in right, left, up and bottom direction. It looks like it is not working with hexagonal tiles:

Here I am toggling that option and it works great on left and right sides but not on diagonal:
https://i.imgur.com/Ir2yuaY.gif

I can kinda cheat and make it work by changing separation and margins (chop 1px from tile). But that creates artefacts. I am pretty sure that tiles are 120x140 with 2px separation between rows and columns:
![obraz](https://user-images.githubusercontent.com/62170071/211569836-55c6ae46-2616-4065-8271-7c4196ae4711.png)
![obraz](https://user-images.githubusercontent.com/62170071/211569868-b0a90b49-1fb1-49bb-bdef-978cbe31f8d1.png)

### Steps to reproduce

Try to use kenney hexagonal tiles in godot 4
https://www.kenney.nl/assets/hexagon-pack

### Minimal reproduction project

.

**Comment by groud** [comment]
This is expected. The mode does not modify the inside of the tiles themselves. The whole square is drawn even if you have empty spaces around the hexagonal shape. And this will not change, as a lot of user might require drawn areas that are bigger than the grid size (like, if you have a building in a top-down game, it might be taller than the base hexagon), and we thus cannot modify the tile itself. So basilcally we cannot touch the diagonal transitions as they are inside the drawn square, and not on its borders.

Basically, this option is only here to avoid issues due to the rectangular sampling of the TileSet, which might suffer from precision issues. There's no real solution for diagonals besides making sure the transitions are pixel-perfect. To do so, you might disable filtering on the TileMap. 

And another solution might be to extend the diagonal borders a bit, but not sure it will lead to a clean enough look.

Closing as this cannot be fixed.
